File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program17.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return the count of odd digits.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
using namespace std;

class Display
{
	private:
		int iNo;

	public:

		Display(int iNo)
		{
			this->iNo = iNo;
		}	

		int CountOdd()
		{	
			int iDigit = 0,iCnt=0;

			if(iNo<0)
			{
				iNo = -iNo;
			}

			while(iNo>0)
			{
				iDigit = iNo%10;
				if(iDigit%2!=0)
				{
					iCnt++;
				}
				iNo = iNo/10;
			}

			return iCnt;
		}

};

int main()
{
	int iValue=0,iRet=0;

	cout<<"Enter Number: ";
	cin>>iValue;

	Display *dObj = new Display(iValue);
	iRet = dObj->CountOdd();

	cout<<"Odd count is "<<iRet<<"\n";
	return 0;
}

/*
OUTPUT:

Enter Number: 2395
Odd count is 3

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program17.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		a program which accept number from user and return the count of odd digits.
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.*;

class Solution
{
	public int iValue;

	public Solution(int iValue)
	{
		this.iValue = iValue;
	}

	public int CountOdd()
	{
                      	
		int iCnt=0;
		while(iValue!=0)
		{
			int iDigit = iValue%10;
			if(iDigit%2!=0)
			{
				iCnt++;
			}
			iValue = iValue/10;
		}        
		return iCnt;
	
	}
}

class Program
{
	public static void main(String[] args)
	{
		Scanner sObj = new Scanner(System.in);

		int iRet=0;
		System.out.println("Enter number: ");
		int iNo = sObj.nextInt();

		Solution obj = new Solution(iNo);
		iRet = obj.CountOdd();
	
		System.out.println("Odd count is: "+iRet);
	}
}

/*
OUTPUT:

Enter number: 
2395
Odd count is: 3

*/
